File: Data Structures & Algorithms/longest-repeating-substring-with-replacement/submission-1.java
class Solution {
    public int characterReplacement(String s, int k) {

    Map <Character, Integer> letter = new HashMap<>();

    int l = 0;
    int maxf = 0;
    int res = 0;

for (int r = 0; r < s.length(); r++) {
    letter.put(s.charAt(r), letter.getOrDefault(s.charAt(r), 0) + 1); 
    
    maxf = Math.max(maxf, letter.get(s.charAt(r)));

    while ((r-l+1) - maxf > k) {
        letter.put(s.charAt(l), letter.get(s.charAt(l)) -1);
        l++;
    }
    res = Math.max(res, r - l + 1);
}
return res;

}

}

File: Data Structures & Algorithms/max-water-container/submission-1.py
class Solution:
    def maxArea(self, heights: List[int]) -> int:
    
        l = 0
        r = len(heights) - 1
        area = 0
        while (l<r):
            if heights[l] < heights[r]:
                area = max(area, (r-l)*(heights[l]))
            else:
                area = max(area, (r-l)*(heights[r]))
            
            if heights[l]<heights[r]:
                l = l+1
            else: 
                r = r-1
            
        return area